### PR TITLE
support to manage cluster by proxy 

### LINKED ISSUE #257: karmada supports proxy to access apiserver
**What would you like to be added**:
karmada supports proxy to access apiserver. For example. "proxy-url" parameter in kubeconfig.
**Why is this needed**:
In edge-computing scenarios, when karmada is in cloud , apiserver is in edge. Karmada needs proxy to access apiserver

### PR DESCRIPTION
**Pull request by liufen90**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
support to manage cluster by proxy

**Which issue(s) this PR fixes**:
Fixes #257 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
"NONE"

```release-note

```

### CODE REVIEW

**Comment by kevin-wangzefeng** [comment]
lint is failling, please fix it.

@RainbowMango It seems lint failure details not shown in the ci log, are we missing any configuration?

**Comment by RainbowMango** [comment]
May be `golangci-lint` output format restrict to `github-actions`:
```
/home/runner/golangci-lint-1.32.2-linux-amd64/golangci-lint run --out-format=github-actions
```

We can add more formats, like `colored-line-number`.

**Comment by RainbowMango** [comment]
Seems golangci-lint not support more than one output format:
```
# golangci-lint run --out-format="colored-line-number|github-actions"
ERRO Running error: unknown output format colored-line-number|github-actions
```

For now, action workflow runs with `--out-format=github-actions`, and local verification(`make verify`) runs with `--out-format=colored-line-number`. 

How about both GitHub action and local use `make verify`?

File: pkg/karmadactl/join.go [hunk 1]
```diff
@@ -266,6 +266,15 @@ func RunJoin(cmdOut io.Writer, karmadaConfig KarmadaConfig, opts CommandJoinOpti
 		clusterObj.Spec.InsecureSkipTLSVerification = true
 	}
 
+	if clusterConfig.Proxy != nil {
+		url, err := clusterConfig.Proxy(nil)
+		if err != nil {
+			klog.Errorf("clusterConfig.Proxy error, %v", err)
+			return err
+		}
+		clusterObj.Spec.ProxyURL = url.String()
```
Comment (RainbowMango) [inline pkg/karmadactl/join.go:275]: Can we presume that the `url` never be `nil`?

**Comment by RainbowMango** [comment]
Just a question, otherwise LGTM.

**Pull request merged**